cli/build-public-library.js

```diff
@@ -32,81 +29,69 @@ function cleanAll() {
   );
 }
 
-function cleanRuntime() {
-  rimraf.sync(skyPagesConfigUtil.spaPath('dist', 'runtime'));
-}
+function stageSourceFiles() {
+  // Copy everything in the `public` folder.
+  fs.copySync(
+    skyPagesConfigUtil.spaPath('src/app/public'),
+    skyPagesConfigUtil.spaPathTemp()
+  );
+
+  // Copy `package.json`.
+  fs.copySync(
+    skyPagesConfigUtil.spaPath('package.json'),
+    skyPagesConfigUtil.spaPathTemp('package.json')
+  );
 
-function getEntryPointFiles() {
-  const files = [
-    skyPagesConfigUtil.spaPathTemp('index.ts')
+  // Copy specific static files.
+  const pathsToCopy = [
+    ['README.md'],
+    ['CHANGELOG.md'],
+    ['src', 'assets']
   ];
 
-  const testingPath = skyPagesConfigUtil.spaPathTemp('testing', 'index.ts');
-  if (fs.existsSync(testingPath)) {
-    files.push(testingPath);
-  }
+  pathsToCopy.forEach(pathArr => {
+    const sourcePath = skyPagesConfigUtil.spaPath(...pathArr);
+    if (fs.existsSync(sourcePath)) {
+      fs.copySync(
+        sourcePath,
+        skyPagesConfigUtil.spaPath('dist', ...pathArr)
+      );
+    } else {
+      logger.warn(`File(s) not found: ${sourcePath}`);
+    }
+  });
+}
 
-  return files;
+function cleanRuntime() {
+  rimraf.sync(skyPagesConfigUtil.spaPath('dist', 'runtime'));
 }
 
 function writeTSConfig() {
+  // TODO: Enable template checking in a breaking change.
   const config = {
-    'compilerOptions': {
-      'target': 'es5',
-      'module': 'es2015',
-      'moduleResolution': 'node',
-      'emitDecoratorMetadata': true,
-      'experimentalDecorators': true,
-      'allowSyntheticDefaultImports': true,
-      'sourceMap': true,
-      'importHelpers': true,
-      'noEmitHelpers': true,
-      'noImplicitAny': true,
-      'declaration': true,
-      'skipLibCheck': true,
-      'inlineSources': true,
-      'lib': [
-        'dom',
-        'es6'
-      ],
-      'typeRoots': [
-        skyPagesConfigUtil.spaPath('node_modules', '@types')
-      ],
-      'outDir': skyPagesConfigUtil.spaPath('dist'),
-      'rootDir': skyPagesConfigUtil.spaPathTemp(),
-      'baseUrl': '.',
-      'paths': {
-        '@skyux-sdk/builder/*': [
-          '*'
-        ],
-        '.skypageslocales/*': [
-          '../src/assets/locales/*'
-        ]
-      }
+    extends: skyPagesConfigUtil.spaPath(
+      'node_modules/ng-packagr/lib/ts/conf/tsconfig.ngc.json'
+    ),
+    compilerOptions: {
+      lib: ['dom', 'es6']
     },
-    'files': getEntryPointFiles(),
-    'exclude': [
-      'node_modules',
-      '**/*.spec.ts',
-      '**/*.e2e-spec.ts',
-      '**/*.pact-spec.ts'
-    ],
-    'angularCompilerOptions': {
-      'annotateForClosureCompiler': true,
-      'fullTemplateTypeCheck': false,
-      'skipTemplateCodegen': true,
-      'strictMetadataEmit': true,
-      'strictInjectionParameters': true,
-      'enableResourceInlining': true
-    },
-    'compileOnSave': false
+    angularCompilerOptions: {
+      fullTemplateTypeCheck: false,
```

Comment:
This setting will throw build errors if a property in your HTML template does not match a public property on the component's class. It's super useful, but this probably constitutes a breaking change because many builds could possibly fail if this were activated (note: this setting is disabled in our current tsconfig.json file).

Reply:
Since this will be added to RC, we should be good to include template checking.